pkg/hintrunner/zero/zerohint_others.go

```diff
@@ -144,3 +147,146 @@ func createMemEnterScopeHinter(resolver hintReferenceResolver, memset bool) (hin
+//   - `nElms` is the number of elements in the array
+//   - `key` is the given key that acts a threshold
+//   - `index` is the result, i.e., the index of the first element greater or equal to the given key
+func newSearchSortedLowerHint(arrayPtr, elmSize, nElms, key, index hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SearchSortedLower",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> array_ptr = ids.array_ptr
+			//> elm_size = ids.elm_size
+			//> assert isinstance(elm_size, int) and elm_size > 0, \
+			//> 	f'Invalid value for elm_size. Got: {elm_size}.'
+			//>
+			//> n_elms = ids.n_elms
+			//> assert isinstance(n_elms, int) and n_elms >= 0, \
+			//> 	f'Invalid value for n_elms. Got: {n_elms}.'
+			//> if '__find_element_max_size' in globals():
+			//> 	assert n_elms <= __find_element_max_size, \
+			//> 		f'find_element() can only be used with n_elms<={__find_element_max_size}. ' \
+			//> 		f'Got: n_elms={n_elms}.'
+			//>
+			//> for i in range(n_elms):
+			//> 	if memory[array_ptr + elm_size * i] >= ids.key:
+			//> 		ids.index = i
+			//> 		break
+			//> else:
+			//> 	ids.index = n_elms
+
+			//> array_ptr = ids.array_ptr
+			arrayPtr, err := hinter.ResolveAsUint64(vm, arrayPtr)
+			if err != nil {
+				return err
+			}
+
+			//> elm_size = ids.elm_size
+			elmSize, err := hinter.ResolveAsUint64(vm, elmSize)
+			if err != nil {
+				return err
+			}
+
+			//> assert isinstance(elm_size, int) and elm_size > 0, \
+			//> 	f'Invalid value for elm_size. Got: {elm_size}.'
+			if elmSize == 0 {
+				return fmt.Errorf("invalid value for elm_size. Got: %v", elmSize)
+			}
+
+			//> n_elms = ids.n_elms
+			//> assert isinstance(n_elms, int) and n_elms >= 0, \
+			//> 	f'Invalid value for n_elms. Got: {n_elms}.'
+			nElms, err := hinter.ResolveAsFelt(vm, nElms)
+			if err != nil {
+				return err
+			}
+
+			//> if '__find_element_max_size' in globals():
+			//> 	assert n_elms <= __find_element_max_size, \
+			//> 		f'find_element() can only be used with n_elms<={__find_element_max_size}. ' \
+			//> 		f'Got: n_elms={n_elms}.'
+			elementMaxSize, err := ctx.ScopeManager.GetVariableValue("__find_element_max_size")
+			if err == nil {
+				elementMaxSizeFelt, ok := elementMaxSize.(fp.Element)
+				if !ok {
+					return fmt.Errorf("failed obtaining the variable: __find_element_max_size")
+				}
+
+				if !utils.FeltLe(nElms, &elementMaxSizeFelt) {
+					return fmt.Errorf("find_element() can only be used with n_elms<=%v. Got: n_elms=%v", elementMaxSizeFelt, nElms)
+				}
+			}
+
+			key, err := hinter.ResolveAsFelt(vm, key)
+			if err != nil {
+				return err
+			}
+
+			indexAddr, err := index.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			nElemsRange := nElms.Uint64()
+
+			//> for i in range(n_elms):
+			for i := uint64(0); i < nElemsRange; i++ {
+				//> 	if memory[array_ptr + elm_size * i] >= ids.key:
+				index := arrayPtr + (elmSize * i)
```

Comment:
If you take arrayPtr as an address as mentioned above, then you'll probably need to change the Add operation here.
Btw, one idea here. Instead of calculating the multiplication each time, you could instead update the value of index in each iteration by adding elmSize to it. You'd have to start `index = arrPtr` and do the update at the end of the loop so it has the correct value for the next iteration. Wdyt?

Reply:
updated! better like this for sure, thanks!